.github/workflows/litex_sim.yml

```diff
@@ -0,0 +1,130 @@
+        run: |
+          sudo apt update
+          sudo apt install python3-pip python3-venv \
+            verilator libevent-dev libjson-c-dev libz-dev
+
+      # Uses a custom RISCV toolchain containing the required headers. This
+      # follows the libtock-c GitHub actions definition.
+      - name: Setup RISC-V GCC toolchain
+        run: |
+          pushd $HOME
+          wget -q http://cs.virginia.edu/~bjc8c/archive/gcc-riscv64-unknown-elf-8.3.0-ubuntu.zip
+          unzip gcc-riscv64-unknown-elf-8.3.0-ubuntu.zip
+          echo "$HOME/gcc-riscv64-unknown-elf-8.3.0-ubuntu/bin" >> $GITHUB_PATH
+          popd
+
+      # Unfortunately we also need to provision an ARM toolchain. While we don't
+      # want to actively use it, some apps override the TOCK_TARGETS in their
+      # Makefile and thus do build for cortex-m platforms. This breaks badly
+      # when no matching toolchain is installed.
+      - name: Setup ARM GCC toolchain
+        uses: fiam/arm-none-eabi-gcc@v1
+        with:
+          release: '9-2019-q4' # The arm-none-eabi-gcc release to use.
+
+      # Setup a Rust toolchain (pulls version from rust-toolchain file)
+      - name: Setup Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          components: rustfmt, clippy
+
+      # Install tockloader, which is used to prepare binaries with userspace
+      # applications. TODO: switch to a released version with the flash file
+      # support included.
+      - name: Install tockloader
+        run: |
+          pip3 install "git+https://github.com/tock/tockloader@master#egg=tockloader"
+
+      # Clone tock-litex support repository under ./tock-litex, check out the
+      # targeted release. TODO: switch to a release with the Python
+      # requirements.txt included.
+      - name: Checkout the tock-litex repository
+        uses: actions/checkout@v2
+        with:
+          repository: lschuermann/tock-litex
+          ref: master
+          path: tock-litex
+
+      # Install all of the required Python packages from the tock-litex'
+      # requirements.txt file
+      - name: Install Python packages pinned by the tock-litex revision
+        run: |
+          pushd tock-litex
+          pip3 install -r requirements.txt
+          popd
+
+      # Build the LiteX simulator Tock kernel. This kernel is never touched, the
+      # litex-ci-runner will use its own temporary flash files.
+      - name: Build the LiteX simulator Tock kernel
+        run: |
+          pushd boards/litex/sim
+          make
+          popd
+
+      # TODO: Should we pin a specific targeted libtock-c version for all CI in
+      # Tock or always refer to the master? It'd probably be good if CI is
+      # somewhat reproducible and doesn't change in behavior because of changes
+      # to another repo.
+      - name: Checkout libtock-c
+        uses: actions/checkout@v2
+        with:
+          repository: tock/libtock-c
+          ref: master
```

Comment:
I think it would be ok if different CI passes uses different libtock-c commits. Anyone adding a new test workflow might not want to verify that all other test workflows work on the same libtock-c, or fix (and wait for the fixes to be merged) libtock-c to make sure all can use the same libtock-c commit.

But, can we at least make the file not hidden and put it in the .github folder? Anything we add to the root pushes the readme farther down the fold on the github page.